src/screens/VolumeCalibrationScreen.tsx

```diff
@@ -0,0 +1,231 @@
+      }
+    }, 1000)
+  }
+
+  const startCalibration = async () => {
+    // Show countdown timer
+    setStage(1)
+
+    // Trigger Countdown
+    const testCurrentVolume = () => {
+      setCountdown(3)
+      decreaseCountdown(countdown, async () => {
+        // Play Sound
+        await playStimuli()
+
+        // Ask participant what they thought
+        Alert.alert(
+          'How was that?',
+          'Is the volume of the sound uncomfortable?',
+          [
+            {
+              text: 'Yes',
+              onPress: () => {
+                // Switch to volume rating
+                setStage(2)
+              },
+            },
+            {
+              text: 'No',
+              onPress: () => {
+                // If reached max volume
+                if (volume.current >= 1.0) {
+                  setStage(2)
+                } else {
+                  // Increase volume and play again
+                  volume.current = +(volume.current + 0.1).toFixed(2)
+                  testCurrentVolume()
+                }
+              },
+            },
+            {
+              text: 'Not Sure',
+              onPress: () => {
+                // Restart countdown
+                testCurrentVolume()
+              },
+            },
+          ],
+          { cancelable: false },
+        )
+      })
+    }
+
+    // Run volume test
+    testCurrentVolume()
+  }
+
+  const validateRating = () => {
+    // Respond if no rating selected
+    if (volumeRating === undefined) {
+      Alert.alert(
+        'Rate the volume',
+        'Please use the rating scale to describe how you felt about the played sound',
+      )
+    }
+
+    // User finds volume painful, let's verify they want to continue
+    else if (volumeRating === 9) {
+      Alert.alert(
+        'Are you sure',
+        'You may hear sounds at this volume multiple times during the experiment. Do you think you can tolerate this volume?',
+        [
+          {
+            text: 'Yes',
+            onPress: () => onFinishCalibration(volume.current),
+          },
+          {
+            text: 'No',
+            onPress: () => setStage(3),
+          },
+        ],
+        { cancelable: false },
+      )
+    }
+
+    // See if rating is beyond threshold
+    else if (volumeRating > 5) {
+      onFinishCalibration(volume.current)
+    }
+  }
+
+  const resetCalibration = () => {
+    volume.current = 0.5
+    setVolumeRating(undefined)
+    startCalibration()
+  }
+
+  return (
+    <Box
+      flex={1}
+      alignItems="center"
+      pt={24}
+      px={8}
+      backgroundColor="greenPrimary"
+    >
+      <Entypo name="sound" size={100} color={palette.darkGrey} />
+
+      <Text mt={8} variant="caption" textAlign="center">
+        For this experiment it is important that the volume of the sound is
+        uncomfortable but not painful.
+      </Text>
+
+      {/* Instruction Stage */}
+      {stage == 0 && (
```

Comment:
Suggestion: The stages would probably be easier to understand if they were named entities instead of integers. Like a string, or even just a global variable assigned to the numbers. Not a huge deal but I think it would help.